libcaf_core/caf/intrusive/forward_iterator.hpp

```diff
@@ -28,9 +28,9 @@ class forward_iterator {
 
   using const_reference = const value_type&;
 
-  using node_type = typename std::conditional<std::is_const<T>::value,
-                                              const typename T::node_type,
-                                              typename T::node_type>::type;
+  using node_type =
+    typename std::conditional<std::is_const_v<T>, const typename T::node_type,
+                              typename T::node_type>::type;
```

Comment:
```suggestion
    std::conditional_t<std::is_const_v<T>, const typename T::node_type,
                              typename T::node_type>;
```

Reply:
I have observed some other code still uses `::type` instead of `$pred_t<>` syntax. Maybe those can be refactored in a new PR?